### PR TITLE
Add  java8 time api capabilities in DTOs & Model

### LINKED ISSUE #57: Switch to LocalDate from the Java 8 Date Time API
Model and OpenAPI Generator should use the LocalDate.
See https://github.com/spring-petclinic/spring-petclinic-rest/pull/52

### PR DESCRIPTION
**Pull request by alexandre-touret**
Fix #57 

### How does it work ?

* Add java8 time api generation for DTOs using OPENAPIGenerator
* Change model POJOs date to ``LocalDate``

### How was it tested?

This PR was validated running the unit and integration tests

### CODE REVIEW

File: src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcPetRowMapper.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public JdbcPet mapRow(ResultSet rs, int rownum) throws SQLException {
         pet.setId(rs.getInt("pets_id"));
         pet.setName(rs.getString("name"));
         Date birthDate = rs.getDate("birth_date");
-        pet.setBirthDate(new Date(birthDate.getTime()));
+        pet.setBirthDate(new java.sql.Date(birthDate.getTime()).toLocalDate());
```
Comment (arey) [inline src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcPetRowMapper.java:36]: I think we could avoid using the `java.sql.Date` type with something like this:
```
pet.setBirthDate(rs.getObject("birth_date", LocalDate.class));
```

File: src/main/java/org/springframework/samples/petclinic/model/Pet.java [hunk 2]
```diff
@@ -34,10 +35,9 @@
 @Table(name = "pets")
 public class Pet extends NamedEntity {
 
-    @Column(name = "birth_date")
-    @Temporal(TemporalType.DATE)
+    @Column(name = "birth_date", columnDefinition = "DATE")
     @DateTimeFormat(pattern = "yyyy/MM/dd")
```
Comment (arey) [inline src/main/java/org/springframework/samples/petclinic/model/Pet.java:39]: By the way, I'm not sure that `@DateTimeFormat` still have an interest.

**Comment by alexandre-touret** [comment]
Hi @arey 
I fixed the issues you pointed out.

**Comment by arey** [comment]
Thank you @alexandre-touret Could you check why code coverage has decreased?

**Comment by alexandre-touret** [comment]
Hi @arey 
I just checked and reproduced this issue on my laptop.

It fails now because in the github action definition, it's now 

```yaml
      - name: Build with Maven
        run: mvn -B install --file pom.xml
```

instead of 

```yaml
      - name: Build with Maven
        run: mvn -B package --file pom.xml
```

The install step checks the jacoco report and fails the build if the coverage rate is not enough.

**Comment by arey** [comment]
So you have to add some lines of test or decrease the limit of the code coverage.

**Pull request merged**